backend/icpy/services/filesystem_service.py

```
@@ -440,6 +440,33 @@ def __init__(self, root_path: str = None, max_file_size: int = 100 * 1024 * 1024
         # Track directories explicitly created via service to suppress duplicate
         # watchdog-created events (API create -> publish + watchdog on_created)
         self._recent_created_dirs: Dict[str, float] = {}
+
+        # In-memory caches (hot paths)
+        # Configurable via env: FS_INFO_CACHE_SIZE, FS_READ_CACHE_SIZE
+        try:
+            self._info_cache_max = int(os.getenv('FS_INFO_CACHE_SIZE', '1024'))
+        except Exception:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)